encord/client.py

```diff
@@ -504,10 +506,32 @@ def add_private_data_to_dataset(
         else:
             raise ValueError(f"Type [{type(private_files)}] of argument private_files is not supported")
 
-        payload = {"files": files, "integration_id": integration_id, "ignore_errors": ignore_errors}
-        response = self._querier.basic_setter(DatasetData, self._config.resource_id, payload=payload)
+        process_hash = self._querier.basic_setter(
+            DatasetDataLongPolling,
+            self._config.resource_id,
+            payload={
+                "files": files,
+                "integration_id": integration_id,
+                "ignore_errors": ignore_errors,
+            },
+        )["process_hash"]
+
+        for _ in range(10):
+            polling_response = self._querier.basic_getter(
+                DatasetDataLongPolling,
+                self._config.resource_id,
+                payload={
+                    "process_hash": process_hash,
+                    "ignore_errors": ignore_errors,
```

Comment:
why do we have to pass `ignore_errors` once in the get and once in the post?

Reply:
+1, actually.

Totally missed that issue when looking at the BE code

Reply:
Because we depend on ignore_errors both in GET and POST, 
we could hardcode it in either GET or POST but is it ok?, imo not.

ignore_errors in POST determines if we want to silence errors in Kubeflow component doing finalisation step and only report them in GET request (if we want)

ignore_errors in GET determines if we want to raise Exception in GET request, if there were errors in kubeflow components.

Please look at implementation of both entrypoints to see why I think it's required and ok to have ignore_errors 2x.

Reply:
I'm confused here. I thought this would be exactly the flag that we set in the UI of "ignore individual errors" (I would also call it this way actually). Is it not the case?

I think we're leaking some implementation detail. Sure, we can hide things in the SDK, but at the API level someone should not care about the errors "doing finalisation step". Maybe I'm missing something though, but seems not great to me.